Lint fails on angular buildable libraries

### Current Behavior

Linting a angular buildable library if fails with:
```
/.../lint-problem/libs/ng-buildable/package.json
 2:9  error  Parsing error: Unexpected token :
```
Even if is a new library.

### Expected Behavior

Linting an newly created library should not fail.

### GitHub Repo

https://github.com/llucasshenrique/lint-problem

### Steps to Reproduce

1.Create an buildable angular library
2.Run lint as it as target

### Nx Report

```shell
>  NX  Falling back to ts-node for local typescript execution. This may be a little slower.
  - To fix this, ensure @swc-node/register and @swc/core have been installed

 >  NX   Report complete - copy this into the issue template

   Node   : 18.17.1
   OS     : linux-x64
   pnpm   : 8.6.9
   
   nx (global)        : 16.3.2
   nx                 : 16.7.4
   @nx/js             : 16.7.4
   @nx/jest           : 16.7.4
   @nx/linter         : 16.7.4
   @nx/workspace      : 16.7.4
   @nx/angular        : 16.7.4
   @nx/cypress        : 16.7.4
   @nx/eslint-plugin  : 16.7.4
   @nx/web            : 16.7.4
   nx-cloud           : 16.3.0
   typescript         : 5.1.3
```


### Failure Logs

```shell
> nx run ng-buildable:lint


Linting "ng-buildable"...
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.1.0

YOUR TYPESCRIPT VERSION: 5.1.3

Please only submit bug reports when using the officially supported version.

=============

.../lint-problem/libs/ng-buildable/package.json
  2:9  error  Parsing error: Unexpected token :

✖ 1 problem (1 error, 0 warnings)

Lint errors found in the listed files.


 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target lint for project ng-buildable (3s)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```


### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

If fails on new an old buildable libraries, i was migrating old non-buildable libraries to buildable ones and this errors appeared after i've updated the workspace.

--- a/packages/angular/src/generators/add-linting/add-linting.ts
+++ b/packages/angular/src/generators/add-linting/add-linting.ts
@@ -48,6 +48,11 @@ export async function addLintingGenerator(
 
     replaceOverridesInLintConfig(tree, options.projectRoot, [
       {
+        files: ['*.json'],
+        parser: 'jsonc-eslint-parser',
+        rules: {},
+      },
+      {
         files: ['*.ts'],
         ...(hasParserOptions
           ? {

--- a/packages/angular/src/generators/add-linting/lib/add-angular-eslint-dependencies.ts
+++ b/packages/angular/src/generators/add-linting/lib/add-angular-eslint-dependencies.ts
@@ -4,6 +4,8 @@ import { versions } from '../../utils/version-utils';
 
 export function addAngularEsLintDependencies(tree: Tree): GeneratorCallback {
   const angularEslintVersionToInstall = versions(tree).angularEslintVersion;
+  const jsoncEslintParserVersionToInstall =
+    versions(tree).jsoncEslintParserVersion;
   return addDependenciesToPackageJson(
     tree,
     {},
@@ -11,6 +13,7 @@ export function addAngularEsLintDependencies(tree: Tree): GeneratorCallback {
       '@angular-eslint/eslint-plugin': angularEslintVersionToInstall,
       '@angular-eslint/eslint-plugin-template': angularEslintVersionToInstall,
       '@angular-eslint/template-parser': angularEslintVersionToInstall,
+      'jsonc-eslint-parser': jsoncEslintParserVersionToInstall,
     }
   );
 }

--- a/packages/angular/src/generators/application/application.spec.ts
+++ b/packages/angular/src/generators/application/application.spec.ts
@@ -513,6 +513,13 @@ describe('app', () => {
             ],
             "overrides": [
               {
+                "files": [
+                  "*.json",
+                ],
+                "parser": "jsonc-eslint-parser",
+                "rules": {},
+              },
+              {
                 "extends": [
                   "plugin:@nx/angular",
                   "plugin:@angular-eslint/template/process-inline-templates",

--- a/packages/angular/src/generators/library/library.spec.ts
+++ b/packages/angular/src/generators/library/library.spec.ts
@@ -591,6 +591,52 @@ describe('lib', () => {
         expect(tree.exists(path)).toBeTruthy();
       });
 
+      expect(tree.read('my-dir/my-lib/.eslintrc.json', 'utf-8'))
+        .toMatchInlineSnapshot(`
+        "{
+          "extends": ["../../.eslintrc.json"],
+          "ignorePatterns": ["!**/*"],
+          "overrides": [
+            {
+              "files": ["*.json"],
+              "parser": "jsonc-eslint-parser",
+              "rules": {}
+            },
+            {
+              "files": ["*.ts"],
+              "extends": [
+                "plugin:@nx/angular",
+                "plugin:@angular-eslint/template/process-inline-templates"
+              ],
+              "rules": {
+                "@angular-eslint/directive-selector": [
+                  "error",
+                  {
+                    "type": "attribute",
+                    "prefix": "proj",
+                    "style": "camelCase"
+                  }
+                ],
+                "@angular-eslint/component-selector": [
+                  "error",
+                  {
+                    "type": "element",
+                    "prefix": "proj",
+                    "style": "kebab-case"
+                  }
+                ]
+              }
+            },
+            {
+              "files": ["*.html"],
+              "extends": ["plugin:@nx/angular-template"],
+              "rules": {}
+            }
+          ]
+        }
+        "
+      `);
+
       // Make sure these have properties
       [
         {
@@ -1111,6 +1157,13 @@ describe('lib', () => {
               "!**/*",
             ],
             "overrides": [
+              {
+                "files": [
+                  "*.json",
+                ],
+                "parser": "jsonc-eslint-parser",
+                "rules": {},
+              },
               {
                 "extends": [
                   "plugin:@nx/angular",

--- a/packages/angular/src/utils/backward-compatible-versions.ts
+++ b/packages/angular/src/utils/backward-compatible-versions.ts
@@ -37,6 +37,7 @@ export const backwardCompatibleVersions: Record<
     jestPresetAngularVersion: '~12.2.3',
     typesNodeVersion: '16.11.7',
     jasmineMarblesVersion: '^0.9.2',
+    jsoncEslintParserVersion: '^2.1.0',
   },
   angularV15: {
     angularVersion: '~15.2.0',
@@ -64,5 +65,6 @@ export const backwardCompatibleVersions: Record<
     jestPresetAngularVersion: '~13.0.0',
     typesNodeVersion: '16.11.7',
     jasmineMarblesVersion: '^0.9.2',
+    jsoncEslintParserVersion: '^2.1.0',
   },
 };

--- a/packages/angular/src/utils/versions.ts
+++ b/packages/angular/src/utils/versions.ts
@@ -28,3 +28,5 @@ export const tsNodeVersion = '10.9.1';
 export const jestPresetAngularVersion = '~13.1.0';
 export const typesNodeVersion = '16.11.7';
 export const jasmineMarblesVersion = '^0.9.2';
+
+export const jsoncEslintParserVersion = '^2.1.0';

